Formula/buildifier.rb

```diff
@@ -13,11 +13,96 @@ class Buildifier < Formula
     sha256 cellar: :any_skip_relocation, high_sierra: "5b54427b4bd78b0b3dccfd66b8003a52aab3b4d1a7683586b66d6a6c835c0b4b"
   end
 
-  depends_on "bazelisk" => :build
+  depends_on "go" => :build
 
   def install
-    system "bazelisk", "build", "--config=release", "buildifier:buildifier"
-    bin.install "bazel-bin/buildifier/darwin_amd64_stripped/buildifier"
+    ENV["GOPATH"] = buildpath
+    ENV["GO111MODULE"] = "auto"
+    commit = Utils.git_short_head
+    ldflags = ["-X main.buildVersion=#{version}",
+               "-X main.buildScmRevision=#{commit}"]
+    dir = buildpath/"buildifier"
+    (dir/"go.mod").write <<~EOS
+      module github.com/bazelbuild/buildifier
+      go 1.15
+      require (
+        github.com/bazelbuild/buildtools v0.0.0-20210208125002-752270cf2f52
+        github.com/golang/protobuf v1.4.3 // indirect
+        google.golang.org/protobuf v1.25.0 // indirect
+      )
+    EOS
+
+    (dir/"go.sum").write <<~EOS
```

Comment:
Does upstream not have these? We do not want to maintain them in the formula.

Reply:
Upstream doesn't have those as they build it with bazel - which is total overkill and failes to build right now because of reasons that have to do with bazel not buildifier or go

Reply:
If we go from "unmaintainable because of bazel" to "unmaintainable because of hardcoded files" it doesn't really fix anything for brew. In that case we're better off deprecating the formula because it doesn't build.

Reply:
I opened a PR https://github.com/bazelbuild/buildtools/pull/959 on upstream, but the test failed. And I don't know how to fix it because I don't use Bazel at all. Any help?

Reply:
> If we go from "unmaintainable because of bazel" to "unmaintainable because of hardcoded files" it doesn't really fix anything for brew. In that case we're better off deprecating the formula because it doesn't build.  

It's not pretty but it's far from "unmaintainable because of hardcoded files".  
It's version pinned and might not be an issue with the next update when upstream switches to go modules.  
If they don't switch then it's super easy to update.

Reply:
> It's not pretty but it's far from "unmaintainable because of hardcoded files".

I disagree

> be maintained (i.e. the last release wasn’t ages ago, it works without patching on all supported macOS releases and has no outstanding, unpatched security vulnerabilities)

https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

We're essentially opening this rule up and stretching it to it's very limits because
1. It requires heavy patching (adding a new build system)
2. This patching might introduce security vulnerabilities because homebrew isn't going to keep track of all used dependencies.

I don't know if @Homebrew/core disagrees, but I think we need to drop this formula because we can't make it work anymore.